### DISCUSSION TITLE
Support different URL Formats in the docstring parser.

### DISCUSSION THREAD

**Discussion by shadycuz** [microsoft/pylance-release#2668] [Enhancement]
I'm not sure if label is the correct terminology.  But you know when you have [some word](https://google.com) that is backed by a URL. 

pylance seems to support URLs but not the label kind, I have tried different variations including [google](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) and [reST](https://docs.typo3.org/m/typo3/docs-how-to-document/master/en-us/WritingReST/Hyperlinks.html) style links. 

I opened a discussion because I wasn't sure if I was just doing it wrong or maybe its the responsibility of the python extension and not the language server.

![image](https://user-images.githubusercontent.com/17090999/114852325-47fec680-9e00-11eb-8ea3-df5ee2e92607.png)

Docstring:

```
"""AWS Intrinsic functions.

This module contains the logic to render both AWS `Intrinsic function`_
and https://google.com

by the `Google Python Style Guide`_.
.. _Intrinsic function:
   https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html
.. _Google Python Style Guide:
   http://google.github.io/styleguide/pyguide.html
"""
```

_Originally posted by @shadycuz in https://github.com/microsoft/pylance-release/discussions/1160_

**Comment by shadycuz** [comment]
Seems like it might be better to move this to pyright? That has the docstring parser it looks [like](https://github.com/microsoft/pyright/blob/master/packages/pyright-internal/src/analyzer/docStringConversion.ts).

**Comment by jakebailey** [comment]
The pyright repo is largely only for type checking issues. A copy of the code lives there but we manage all editor integrations here; the type checker doesn't need docstrings.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
